core/include/vecmem/containers/impl/jagged_vector_buffer.ipp

```diff
@@ -116,16 +126,25 @@ jagged_vector_buffer<TYPE>::jagged_vector_buffer(
     // Set up the vecmem::vector_view objects in the host accessible memory.
     std::ptrdiff_t ptrdiff = 0;
     for (std::size_t i = 0; i < capacities.size(); ++i) {
-        if (header_ptr != nullptr) {
-            new (base_type::host_ptr() + i) value_type(
-                static_cast<typename value_type::size_type>(capacities[i]),
-                &(header_ptr[i]), data_ptr + ptrdiff);
+        if (capacities[i] != 0u) {
+            if (header_ptr != nullptr) {
+                new (base_type::host_ptr() + i) value_type(
+                    static_cast<typename value_type::size_type>(capacities[i]),
+                    &(header_ptr[i]), data_ptr + ptrdiff);
+            } else {
+                new (base_type::host_ptr() + i) value_type(
+                    static_cast<typename value_type::size_type>(capacities[i]),
+                    data_ptr + ptrdiff);
+            }
+            ptrdiff += capacities[i];
         } else {
-            new (base_type::host_ptr() + i) value_type(
-                static_cast<typename value_type::size_type>(capacities[i]),
-                data_ptr + ptrdiff);
+            if (header_ptr != nullptr) {
+                new (base_type::host_ptr() + i)
+                    value_type(0, &(header_ptr[i]), nullptr);
+            } else {
+                new (base_type::host_ptr() + i) value_type(0, nullptr);
+            }
```

Comment:
It's not, like I mentioned here before, the whole code works fine without any changes since #232 was merged in. I can just leave in the 1st change of not allocating device memory for `fixed_size` buffers with 0 total size.
My point was to try to match the innards of the `jagged_vector_buffer` to what the `vector_buffer` does with using `nullptr` for emptyness, and also trying to reduce the ammount of unnecessary calculations being made in these cases, altough I guess this might not be achieving so much anyways.